src/summary/history/conditions-card.component.tsx

```diff
@@ -0,0 +1,65 @@
+import React from "react";
+import style from "./conditions-card-style.css";
+import dayjs from "dayjs";
+import SummaryCard from "../cards/summary-card.component";
+import SummaryCardRow from "../cards/summary-card-row.component";
+import SummaryCardRowContent from "../cards/summary-card-row-content.component";
+import { match } from "react-router";
+import { performPatientConditionSearch } from "./conditions.resource";
+
+export default function ConditionsCard(props: ConditionsCardProps) {
+  const [patientConditions, setPatientConditions] = React.useState(null);
+
+  React.useEffect(() => {
+    const abortController = new AbortController();
+    performPatientConditionSearch(
+      props.currentPatient.identifier[0].value,
+      abortController
+    )
+      .then(condition => setPatientConditions(condition))
+      .catch(error => {
+        throw error;
```

Comment:
throwing an error inside of `.catch()` is a no-op and does not do what you think. This is why I've been encouraging `setTimeout(() => {throw err})`. The reason it is a no-op is because `.catch()` changes the promise from `rejected` to `fulfilled` status, but then throwing inside of the catch just puts the promise back into `rejected` status.

Luckily, openmrs-esm-error-handling has a nice util function for this, though! The following is **the pattern we should follow from now on:**

```js
import { createErrorHandler } from '@openmrs/esm-error-handling';

doThing()
.catch(createErrorHandler())
```

Reply:
seems this functionality hasn't been published to npmjs

Reply:
It doesn't need to be published to npm to work in the browser. It is an in-browser dependency that [is a webpack external](https://github.com/openmrs/openmrs-esm-patient-chart/blob/c3aaf15e4facfc3ec0a77479c01e2c309dee8cdc/webpack.config.js#L50).

And for tests, it's best just to mock external modules. See [example](https://github.com/openmrs/openmrs-esm-patient-chart/blob/master/__mocks__/openmrs-esm-api.mock.tsx). You won't be able to use the published version of esm-error-handling in your tests anyway.

All of that said, I published [esm-error-handling@1.1.0](https://github.com/openmrs/openmrs-esm-error-handling/releases/tag/v1.1.0). It won't help you with this PR, but it needed to be done 😄

Reply:
worked on this thank you for this feedback